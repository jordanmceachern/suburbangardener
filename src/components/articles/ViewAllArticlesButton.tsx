import Link from "next/link";

interface ViewAllArticlesButtonProps {
  className?: string;
  variant?: "primary" | "secondary";
  size?: "small" | "medium" | "large";
  text?: string;
  arrowDirection?: "left" | "right";
}

export default function ViewAllArticlesButton({
  className = "",
  variant = "primary",
  size = "medium",
  text = "View All Articles",
  arrowDirection = "right",
}: ViewAllArticlesButtonProps) {
  const baseClasses =
    "font-medium rounded-lg transition-colors inline-flex items-center justify-center";

  const variantClasses = {
    primary:
      "bg-primary-500 hover:bg-primary-600 text-white hover:text-neutral-100",
    secondary:
      "bg-secondary-500 hover:bg-secondary-600 text-white hover:text-neutral-100",
  };

  const sizeClasses = {
    small: "py-2 px-4 text-sm",
    medium: "py-3 px-6 text-base",
    large: "py-4 px-8 text-lg",
  };

  const combinedClasses = `${baseClasses} ${variantClasses[variant]} ${sizeClasses[size]} ${className}`;

  return (
    <Link href="/articles" className={combinedClasses}>
      {arrowDirection === "left" && (
        <svg
          className="w-5 h-5 mr-2"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            d="M15 19l-7-7 7-7"
          />
        </svg>
      )}
      {text}
      {arrowDirection === "right" && (
        <svg
          className="w-5 h-5 ml-2"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={2}
            d="M9 5l7 7-7 7"
          />
        </svg>
      )}
    </Link>
  );
}
