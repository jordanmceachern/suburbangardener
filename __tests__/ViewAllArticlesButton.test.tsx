import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import ViewAllArticlesButton from "@/components/articles/ViewAllArticlesButton";

describe("ViewAllArticlesButton Component", () => {
  it("should match DOM snapshot", () => {
    const { container } = render(<ViewAllArticlesButton />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should maintain CSS classes for button styling", () => {
    const { container } = render(<ViewAllArticlesButton />);

    // Check for button-related classes
    const hasButtonClasses =
      container.innerHTML.includes("button") ||
      container.innerHTML.includes("btn") ||
      container.innerHTML.includes("bg-") ||
      container.innerHTML.includes("hover:") ||
      container.innerHTML.includes("px-") ||
      container.innerHTML.includes("py-");

    expect(hasButtonClasses).toBe(true);
  });
});
