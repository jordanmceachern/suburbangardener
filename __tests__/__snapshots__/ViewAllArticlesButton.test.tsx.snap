// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`ViewAllArticlesButton Component should match DOM snapshot 1`] = `
<a
  class="font-medium rounded-lg transition-colors inline-flex items-center justify-center bg-primary-500 hover:bg-primary-600 text-white hover:text-neutral-100 py-3 px-6 text-base "
  href="/articles"
>
  View All Articles
  <svg
    class="w-5 h-5 ml-2"
    fill="none"
    stroke="currentColor"
    viewBox="0 0 24 24"
  >
    <path
      d="M9 5l7 7-7 7"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
    />
  </svg>
</a>
`;
